File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> m;
        for(auto i :nums)
        {
            if(m.find(i)!=m.end())
            return 1;
            m[i]=i;
        }
        return 0;
        
    }
};

File: Data Structures & Algorithms/is-anagram/submission-4.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        s1=len(s)
        t1=len(t)
        if s1 != t1:
            return False
        d1 = {}
        
        for i in s:
            if i in d1:
                d1[i]+=1
            else:
                d1[i]=1
        for i in t:
            if i in d1:
                d1[i]-=1
            else:
                return False
        for i in d1:
            if d1[i]>0:
                return False
        return True
            

        

        